crates/biome_js_analyze/src/lint/nursery/no_vue_duplicate_keys.rs

```diff
@@ -0,0 +1,193 @@
+    /// in the script tag, they may cause issues in the template since Vue allows direct
+    /// access to these keys.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```vue,expect_diagnostic
+    /// <script>
+    /// export default {
+    ///     props: ['foo'],
+    ///     data() {
+    ///         return {
+    ///             foo: 'bar'
+    ///         };
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    /// ```vue,expect_diagnostic
+    /// <script>
+    /// export default {
+    ///     data() {
+    ///         return {
+    ///             message: 'hello'
+    ///         };
+    ///     },
+    ///     methods: {
+    ///         message() {
+    ///             console.log('duplicate key');
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    /// ```vue,expect_diagnostic
+    /// <script>
+    /// export default {
+    ///     computed: {
+    ///         count() {
+    ///             return this.value * 2;
+    ///         }
+    ///     },
+    ///     methods: {
+    ///         count() {
+    ///             this.value++;
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```vue
+    /// <script>
+    /// export default {
+    ///     props: ['foo'],
+    ///     data() {
+    ///         return {
+    ///             bar: 'baz'
+    ///         };
+    ///     },
+    ///     methods: {
+    ///         handleClick() {
+    ///             console.log('unique key');
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    /// ```vue
+    /// <script>
+    /// export default {
+    ///     computed: {
+    ///         displayMessage() {
+    ///             return this.message.toUpperCase();
+    ///         }
+    ///     },
+    ///     methods: {
+    ///         clearMessage() {
+    ///             this.message = '';
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    pub NoVueDuplicateKeys {
+        version: "next",
+        name: "noVueDuplicateKeys",
+        language: "js",
+        recommended: true,
+        severity: Severity::Error,
+        domains: &[RuleDomain::Vue],
+        sources: &[RuleSource::EslintVueJs("no-dupe-keys").same()],
+    }
+}
+
+impl Rule for NoVueDuplicateKeys {
+    type Query = VueComponentQuery;
+    type State = RuleState;
+    type Signals = Box<[Self::State]>;
+    type Options = NoVueDuplicateKeysOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let Some(component) = VueComponent::from_potential_component(
+            ctx.query(),
+            ctx.model(),
+            ctx.source_type(),
+            ctx.file_path(),
+        ) else {
+            return Box::new([]);
+        };
+
+        let mut key_declarations: FxHashMap<String, Vec<VueDeclaration>> = FxHashMap::default();
```

Comment:
Are the types needed? If not, we should remove them

Reply:
In this case rust fails to detect it automatically.